### PR TITLE
feat: add heartbeat monitoring

### PR DESCRIPTION
**Pull request by tomasbasham**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,7 +3,7 @@ module github.com/StatusCakeDev/terraform-provider-statuscake/v2
 go 1.19
 
 require (
-	github.com/StatusCakeDev/statuscake-go v1.2.0
+	github.com/StatusCakeDev/statuscake-go v1.3.0
```
Comment (tomasbasham) [inline go.mod:6]: Preemptive. Does not exist yet and causing tests to fail

File: go.sum
Comment (bravoecho) [inline go.sum:1]: is this generated?

**Review by bravoecho — APPROVED** [review]

**Review by chrynx — APPROVED** [review]

File: internal/provider/resource_heartbeat_check.go [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+	if err != nil {
+		return diag.FromErr(err)
+	} else if d.HasChange("tags") {
+		body["tags"] = tags
+	}
+
+	log.Printf("[DEBUG] Updating StatusCake heartbeat check with ID: %s", id)
+	log.Printf("[DEBUG] Request body: %+v", body)
+
+	if err := client.UpdateHeartbeatTestWithData(ctx, id, body).Execute(); err != nil {
```
Comment (chrynx) [inline internal/provider/resource_heartbeat_check.go:241]: Noticed on the openapi PR that these are called `Test`

File: internal/provider/resource_heartbeat_check.go [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+	return resourceStatusCakeHeartbeatCheckRead(ctx, d, meta)
+}
+
+func resourceStatusCakeHeartbeatCheckDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*statuscake.Client)
+	id := d.Id()
+
+	log.Printf("[DEBUG] Deleting StatusCake heartbeat check with ID: %s", id)
+
+	if err := client.DeleteHeartbeatTest(ctx, id).Execute(); err != nil {
```
Comment (chrynx) [inline internal/provider/resource_heartbeat_check.go:254]: same here. Could probably do with a full search on the file

File: internal/provider/resource_heartbeat_check.go (hunk 2 above)
Comment (tomasbasham) [reply internal/provider/resource_heartbeat_check.go:241]: This is correct due to current internal naming conventions.

**Pull request merged**